### ISSUE TITLE
Compile instructions are out of date?

### ISSUE THREAD

**Issue by Greatness7** [tabler/tabler-icons#1070]
Hello, I'm wanting to compile the icon font but the instructions seem to be out of date.

The readme says to run `npm run build-iconfont`, but this script is not defined. Maybe it wants `npm run build:webfont`? I am however wanting to get a `ttf` out of it and not a `woff`, it is not clear what command should be ran to do so.

Another issue I had is that the `includeIcons` paremeter specified in `compile-options.json` seems to be ignored. Or perhaps it is only being respected by some parts of the build process and not others. Despite only specifying a small handful of icons in my json, running the `build:webfont` script seems to be executing `fix-outline.py` on every possible icon which takes several hours to complete.

The aforementioned script also had some errors which I had to fix locally, I've submitted a [PR](https://github.com/tabler/tabler-icons/pull/1069) for that part.

**Comment by Greatness7** [comment]
I was ultimately able to get this to compile by working around some of the bugs in the code:

1) Don't be on a Windows system, the build scripts are not compatible despite this not being mentioned anywhere in the README.
2) Fix the python script as above.
3) Fix `icons-webfont/.build/build-outline.mjs` by wrapping the `glob` function in a `promisify`.
4) Move `compile-options.json` to one folder above the *working directory that your script will be ran from*. I had to put it in the `/packages/` folder to be recognized, contrary to what the README suggests. The code responsible for this is [here](https://github.com/tabler/tabler-icons/blob/main/.build/helpers.mjs#L421).
5) Run `npm run build:webfont` (again contradictory to the README).
6) If you're including many icons, expect to wait a long time, likely hours. The build script sequentially spins up a new python instance for every single icon. Doing these in parallel would be a welcome improvement.
7) You will still get some errors and the script will ultimately fail, but this failure happens late enough that the `.ttf` already exists, so I did not care to investigate further.

**Comment by EdgarsJoja** [comment]
Seems this is still not fixed - got the same problems while compiling fonts.
In addition, after fixing already mentioned issues by @Greatness7 (import sys, compile-options.json placement), I see another problem:
```bash
Error: Undefined variable.
   ╷
46 │ .#{$ti-prefix}-123:before { content: $ti-icon-number-123; }
   │                                      ^^^^^^^^^^^^^^^^^^^
   ╵
  dist/tabler-icons-outline.scss 46:38  root stylesheet
 ELIFECYCLE  Command failed with exit code 65.
 ELIFECYCLE  Command failed with exit code 65.
/usr/src/app/tabler-icons/packages/icons-webfont:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @tabler/icons-webfont@3.9.0 build: `pnpm run copy && pnpm run build:prepare && pnpm run build:outline && pnpm run build:webfont && pnpm run build:css`
Exit status 65
```

It looks like `aliases.json` conflict with `compile-options.json` "includedIcons" array. For outlined & filled types it seems to correctly only attempt to generate only specified "includedIcons" values. However, for "all" type it tries to generate merged "includedIcons" and values from `aliases.json` file, even though I don't need those.
Quick semi-fix here is to just add all `aliases.json` values to "includedIcons" array OR remove aliases from `aliases.json` file, i.e.:
```json
{
  "outline": {
  },
  "filled": {
  }
}
```
then build command finished succesfully.

**Comment by manufacturist** [comment]
In my case, with the latest release of [fontforge](https://github.com/fontforge/fontforge/releases/tag/20251009), the solution presented by @Greatness7 no longer works. Icons are fully filled, becoming a mass of a single color.

For anyone reading this, it's easier to just copy the icon SVGs in your project, create your own custom python script that uses fontforge and does the font / CSS generation for you. Also add a mention to this repository in your open source software list if it's public facing.
